### ISSUE TITLE
Filters do not limit to portal scope

### ISSUE THREAD

**Issue by albenson-usgs** [gbif/hosted-portals#187]
I think this may be related to #158 but a different use case. I added `datasetName` to the filter options and I'm expecting to only be able to select datasets that are scoped to the United States but it is searching all datasets in GBIF. Maybe it is not possible to scope the options to the portal but I think this would be what users are expecting.
<img width="859" alt="Capture" src="https://user-images.githubusercontent.com/7586150/135131561-4d56a224-555e-46f8-8dce-0ce0b1d19c11.PNG">

**Comment by MortenHofft** [comment]
yep, that isn't how it works. It isn't how it works on GBIF.org either. It shows all possible datasets. including datasets that have no occurrences.
I agree it would be nice though.

**Comment by MortenHofft** [comment]
Other filters that behave this way
* dataset
* publisher
* scientific name
* country
* basis of record
and more

For some reason it feels more "wrong" in some cases than others.

**Comment by tucotuco** [comment]
@MortenHofft Do you anticipate that it is intractable to implement portal-level scope filters? I think behavior consistency would be important. I also think the user experience would be less confusing if they were in place.

**Comment by MortenHofft** [comment]
I think it is doable, but it requires a fair amount of work for some of them I believe. And some of it would require changes to our index (which isn't something I can do).

I do fully agree that it would be nice for some (like dataset) and have been wondering from before writing my first line of code when someone would request it. 

But I'm not convinced that it should be done for everything. In #158 you also argued that we shouldn't do it for scientific names for example.

And it feels like it could be complicated to do well. Just take a simple basic filter like `basisOfRecord`. Currently it gets the options from a list of enumerations like https://api.gbif.org/v1/enumeration/basic/BasisOfRecord (except in code and shipped with the library to avoid doing 30 calls to get live enumerations)

That would have to be changed to a dynamic loaded version that checked the possible values in the index for a given data scope (only these 4 basisOfRecord values exists in VertNet). But that can change in 2 seconds when new data is published. And so filter options are suddenly a dynamic size. We then get issues with performance and caching. And currently it isn't even async (for performance reasons), so to add that and even do so from a somewhat slow endpoint that cannot really be cached feels like trouble.

I'm sure we can find decent solutions, but it isn't a simple fix

**Comment by MortenHofft** [comment]
Some but not all filters label by portal scope. For some it is simply too computational expensive and slow at this point. I will close this issue, but feel free to create a new one for specific use cases and then we can evaluate our options for that field and case.
